compiler/rustc_const_eval/src/util/validate_scalar_in_layout.rs

```diff
@@ -0,0 +1,30 @@
+use rustc_middle::query::TyCtxtAt;
+use rustc_middle::ty::layout::TyAndLayout;
+use rustc_middle::ty::{self, ScalarInt};
+
+use crate::const_eval::{CanAccessMutGlobal, CheckAlignment, CompileTimeMachine};
+use crate::interpret::{InterpCx, MemoryKind};
+
+pub(crate) fn validate_scalar_in_layout<'tcx>(
+    tcx: TyCtxtAt<'tcx>,
+    scalar: ScalarInt,
+    layout: TyAndLayout<'tcx>,
+) -> bool {
+    let machine = CompileTimeMachine::new(CanAccessMutGlobal::No, CheckAlignment::Error);
+
+    let mut cx = InterpCx::new(tcx.tcx, tcx.span, ty::TypingEnv::fully_monomorphized(), machine);
```

Comment:
If we can assume `fully_monomorphized` here, we might as well just take the type and compute the layout ourselves.